File: app/src/main/java/com/buffalo/controlefinancas/util/PermissionUtil.java
package com.buffalo.controlefinancas.util;

import android.app.Activity;
import android.content.pm.PackageManager;
import android.os.Build;

import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import java.util.ArrayList;
import java.util.List;

public class PermissionUtil {

	public static int REQUEST_CODE_DEFAULT = 1;
	public static int REQUEST_CODE_LOCATION = 2;
    public static int REQUEST_CODE_WPP = 3;
	public static int REQUEST_CODE_PHONE = 4;
	public static int REQUEST_DATA_BASE_UTIL = 4;

	public static boolean checkPermissions(Activity context, String[] permissions, int requestCode) {
		try {
			if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
				return true;
			}
			List<String> permissionsToAsk = new ArrayList<String>();
			for (String permission : permissions) {
				if (ContextCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
					permissionsToAsk.add(permission);
				}
			}
			if (permissionsToAsk.size() > 0) {
				ActivityCompat.requestPermissions(context, permissionsToAsk.toArray(new String[permissionsToAsk.size()]), requestCode);
				return false;
			}
			return true;
		} catch (Exception e) {
			return false;
		}
	}
}

File: app/src/main/java/com/buffalo/controlefinancas/util/PermissionUtil.kt
package com.buffalo.controlefinancas.util

import android.app.Activity
import android.os.Build
import androidx.core.content.ContextCompat
import android.content.pm.PackageManager
import androidx.core.app.ActivityCompat
import java.lang.Exception
import java.util.ArrayList

object PermissionUtil {

    var PERMISSION_WRITE = 1001

    fun checkPermissions(
        context: Activity?,
        permissions: Array<String>,
        requestCode: Int
    ): Boolean {
        return try {
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
                return true
            }
            val permissionsToAsk: MutableList<String> = ArrayList()
            for (permission in permissions) {
                if (ContextCompat.checkSelfPermission(
                        context!!,
                        permission
                    ) != PackageManager.PERMISSION_GRANTED
                ) {
                    permissionsToAsk.add(permission)
                }
            }
            if (permissionsToAsk.size > 0) {
                ActivityCompat.requestPermissions(
                    context!!,
                    permissionsToAsk.toTypedArray(),
                    requestCode
                )
                return false
            }
            true
        } catch (e: Exception) {
            false
        }
    }
}